src/__tests__/lib/containers/table/datasets/DatasetItemsEditor.test.tsx

```diff
@@ -585,4 +590,77 @@ describe('Dataset Items Editor tests', () => {
       ).toBeInTheDocument(),
     )
   })
+
+  describe('SWC-6177 - Gets difference when item is added/updated/changed', () => {
+    it('Shows no changes when same item with same version is added', async () => {
+      // Render dataset editor with item
+      const mockItem = { entityId: mockFileEntity.id!, versionNumber: 3 }
+      const getDatasetHandler = getDatasetHandlerWithItems([mockItem])
+      server.use(getDatasetHandler, successfulUpdateHandler)
+      await renderComponent()
+
+      // Add same item with same version to dataset
+      mockEntityFinderToAddItems([mockFileReference])
+      addItemsViaEntityFinder()
+
+      // Verify toast showing change is not called
+      expect(mockToastFn).not.toBeCalled()
+    })
+
+    it('Shows item has updated when same item with different version is added', async () => {
+      const getDatasetHandler = getDatasetHandlerWithItems([mockDatasetItem])
+      server.use(getDatasetHandler, successfulUpdateHandler)
+      await renderComponent()
+
+      // Add identical item to existing dataset with different version
+      mockEntityFinderToAddItems([mockFileReference])
+      addItemsViaEntityFinder()
+
+      // Verify toast shows no item has been added and 1 has updated
+      expect(mockToastFn).toBeCalledWith(
+        expect.anything(),
+        'info',
+        expect.objectContaining({
+          title: '0 Items added and 1 Item updated to Dataset',
```

Comment:
"updated to Dataset" is a little odd, I think we currently drop it in that case. Recommend one of these two:
```suggestion
          title: '0 Items added and 1 Item updated',
```

```suggestion
          title: '0 Items added to and 1 Item updated in the Dataset',
```